### ISSUE TITLE
Remove duplicated code in test runtime

### ISSUE THREAD

**Issue by DemiMarie-parity** [paritytech/substrate#3113]
The test runtime duplicates much of its code between `std` and `no_std` builds.  I just wasted four or five hours because I had changed the code in one version, but not the other.

We should deduplicate the code so that this does not waste anyone else’s time.

**Comment by bkchr** [comment]
There are reasons we doing this and we try to minimize the amount of duplicate code as much as possible. Everything is wrapped in a cfg_if and nothing is hidden. Based on this, I will close this again.

**Comment by bkchr** [comment]
What you really want is this: https://github.com/paritytech/substrate/issues/1120

Before we did not have any real apis implemented in the `test-runtime`. Just this `TestApi` that is strictly required to be implemented differently on `std` and `no_std` as we use it to test the runtime api implementation itself.

If you look for example at Grandpa, there we implement the `GrandpaTrait` manual to make it mockable. The given issue should simplify this mocking, but yeah until now I did not really had time for it. Sorry!
